engine/execution/scripts/engine.go

```diff
@@ -63,7 +65,14 @@ func (e *Engine) ExecuteScriptAtBlockID(
 	// return early if state with the given state commitment is not in memory
 	// and already purged. This reduces allocations for scripts targeting old blocks.
 	if !e.execState.HasState(stateCommit) {
-		return nil, fmt.Errorf("failed to execute script at block (%s): state commitment not found (%s). this error usually happens if the reference block for this script is not set to a recent block", blockID.String(), hex.EncodeToString(stateCommit[:]))
+		return nil, fmt.Errorf(
+			"failed to execute script at block (%s): %w (%s). "+
+				"this error usually happens if the reference "+
+				"block for this script is not set to a recent block.",
+			blockID.String(),
+			ErrStateCommitmentPruned,
```

Comment:
This is not the only place returning this error, `queryExecutor.ExecuteScript` could also possibly return pruned error, because it's possible state commitment is pruned after this `execState.HasState` check and before the `queryExecutor.ExecuteScript` call. 

In other words, we also need to handle the pruned error from `e.queryExecutor.ExecuteScript`

Reply:
I can handle it in storehouse